src/App.js

```diff
@@ -1,84 +1,97 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./App.css";
 
+import Weather from './components/Weather';
+import Loading from './components/Loading';
+import Forecast from './components/Forecast';
+
+import{cities}  from './utils/cities';
+import{forecastDashboard}  from './utils/index';
+
+
+
 const App = () => {
+
+  /* API  */
+  const MY_API_KEY = process.env.REACT_APP_MY_API_ID;
+  const CURRENT_WEATHER_API = 'https://api.openweathermap.org/data/2.5/weather?units=metric&APPID=';
+  const FORECAST_WEATHER_API = 'https://api.openweathermap.org/data/2.5/forecast?units=metric&APPID=';
+
+  
+  /*useState for weather and for city */
+  const [weather, setWeather] =  useState(null);
+  const [forecast, setForecast] = useState(null);
+  const [city, setCity] =  useState("Bratislava");
+
+
+  /* fetch function to use api and return response with current weather */
+  const fetchWeather = () => {
+    fetch( CURRENT_WEATHER_API + MY_API_KEY + '&q=' + city)
+      .then(response => response.json())
+      .then(json => setWeather(json))
+  };
+
+  /* fetch function to use api and return response with 5 days forecast weather */
+  const fetchForecast = () => {
+    fetch( FORECAST_WEATHER_API + MY_API_KEY + '&q=' + city)
+      .then(response => response.json())
+      .then(json => setForecast(forecastDashboard(json.list)))
+  };
+
+
+  useEffect(() => {
+    fetchWeather();
+  }, [city]);
+
+  useEffect(() => {
+    fetchForecast();
+  }, [city]);
```

Comment:
Tohle je dobrý, ale je potřeba si uvědomit že tyhle 2 operace běží paralelně. Tj kdyby např fetchWeather trval 3s (může se někde seknout), a fetchForecast 500ms, tak se ti UI updatuje "po kouskách". Občas to chceš, ale většinou načítání dat poslouchá tenhle pattern:
- nastav loading stav
- načti z API všechna data která potřebuješ
- nastav potřebný stav a vypni loading

Plus ještě chceš ošetřit případ, kdy někdo rychle přepíná města, rychlejc než se stačí načíst. Chceš ty staré výsledky co dojdou později než nové zahodit;

tj v tomhle případě:
```
useEffect(() => {
   let cancelled = false;

   setWeather(null);
   setForecast(null);

   Promise.all([
      fetch(CURRENT_WEATHER_API + MY_API_KEY + '&q=' + city).then((response) => response.json()),
fetch(FORECAST_WEATHER_API + MY_API_KEY + '&q=' + city)
    .then((response) => response.json()),
   ]).then(([weather, forecast]) => {
        if(!cancelled){
            setWeather(weather);
            setForecast(forecastDashboard(forecast.list));
        }
   });

   return () => {
      cancelled = true;
   }
}, [city])
```
Ale to ti říkám jen pro informaci a abys viděla, že jakékoli programování s nějakými asynchroními věcmi začíná být o dost složitější :)